lib/blocks.php

```diff
@@ -296,3 +296,197 @@ function gutenberg_enqueue_editor_block_styles_assets() {
 	}
 	add_action( 'enqueue_block_editor_assets', 'gutenberg_enqueue_editor_block_styles_assets' );
 }
+
+/**
+ * Renders the classNames and styles for blocks
+ *
+ * @param string $block_content Output of the current block.
+ * @param array $block Block Object.
+ * @return string New block output.
+ */
+function gutenberg_experimental_apply_classnames_and_styles( $block_content, $block ) {
+	// Don't filter template part blocks since we filter the blocks in each template
+	// part individually.
+	if ( 'core/template-part' == $block[ 'blockName' ] ) {
+		return $block_content;
+	}
+
+	// TODO: Check for supports: _experimental* before adding any of this to a block.
+
+	if ( isset( $block['attrs'] ) ) {
+		$colors     = gutenberg_experimental_build_css_colors( $block['attrs'] );
+		$typography = gutenberg_experimental_build_css_typography( $block['attrs'] );
+
+		$extra_classes    = array_merge(
+			$colors['background']['css_classes'],
+			$colors['text']['css_classes'],
```

Comment:
should we merge text and background outputs inside the `gutenberg_experimental_build_css_colors` function directly to simplify things a bit?

Reply:
Definitely (and just did)!  I originally separated them in the post author block PR to apply the text styles to the interior content element separately (was fighting some conflicting style rules at the time).  But it's not necessary anymore. 😄